pwiz_tools/Skyline/Model/Lib/LibKeyIndex.cs

```diff
@@ -162,8 +162,10 @@ public IList<LibraryKey> MergeKeys(LibKeyIndex that)
                 var thatPeptideKey = thatItem.LibraryKey as PeptideLibraryKey;
                 if (thatPeptideKey == null)
                 {
-                    result.Add(thatItem.LibraryKey);
-                    nonPeptideKeySet.Add(thatItem.LibraryKey);
+                    if (nonPeptideKeySet.Add(thatItem.LibraryKey))

```

Comment:
Is this necessary? Are there duplicates in your LibKeyIndex?

Reply:
Not really, no - it's just belt and suspenders since we're copying to a container that doesn't enforce uniqueness.